File: Data Structures & Algorithms/count-paths/submission-0.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        def func(i,j):
            if i<0 or j<0:
                return 0
            if i==0 or j==0:
                return 1
            return func(i-1,j)+func(i,j-1)
        return func(m-1,n-1)
        

File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> umap;
        vector<int> res;
        for(int i=0;i<nums.size();i++)
        {
            int dummy=target-nums[i];
            if(umap.find(dummy)!=umap.end())
            {
                res.push_back(umap[dummy]);
                res.push_back(i);
                return res;
            }
            else
            {
                umap[nums[i]]=i;
            }
        }
        return res;
    }
};
